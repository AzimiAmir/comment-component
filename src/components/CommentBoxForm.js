import React, { Component } from "react";
import moment from 'moment';
import classes from './CommentBoxForm.module.css';

class CommentBoxForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      comment: {
        name: "",
        message: "",
      }
    };
    this.fieldChangeHandler = this.fieldChangeHandler.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  fieldChangeHandler = event => {
    const { value, name } = event.target;
    this.setState({
      ...this.state,
      comment: {
        ...this.state.comment,
        [name]: value
      }
    });
  };
  onSubmit(event) {
    event.preventDefault();
    if (!this.isFormValid()) {
      this.setState({ error: "All fields are required!" });
      return;
    }
    this.setState({ error: "" });
    let { comment } = this.state;
    comment.time = moment().startOf('minute').fromNow();
    this.props.addComment(comment);
    this.setState({
      comment: { ...comment, message: "" }
    });
  }
  isFormValid() {
    return this.state.comment.name !== "" && this.state.comment.message !== "";
  }
  alertError() {
    return this.state.error ? (
      <div className={classes.Alert}>{this.state.error}</div>
    ) : null;
  }
  render() {
    return (
      <React.Fragment>
        <form className={classes.App_Form} onSubmit={this.onSubmit}>
          <div className={classes.App_Form_Text}>
            <input
              type="text"
              name="name"
              onChange={this.fieldChangeHandler}
              value={this.state.comment.name}
              placeholder="Name"
            />
          </div>
          <div className={classes.App_Form_Text}>
            <textarea
              name="message"
              onChange={this.fieldChangeHandler}
              value={this.state.comment.message}
              placeholder="Comment"
            />
          </div>
          <div>
            <button className={classes.Form_Button}>
              Post Comment
            </button>
          </div>
          {this.alertError()}
        </form>
      </React.Fragment>
    );
  }
}

export default CommentBoxForm;