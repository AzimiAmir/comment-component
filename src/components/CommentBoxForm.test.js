import React from 'react'
import {configure, mount} from 'enzyme'
import adapter from 'enzyme-adapter-react-16'
import CommentBoxForm from './CommentBoxForm'

configure({adapter: new adapter()})

describe('<CommentBoxForm />', () => {
  it('should be possible to activate button with click', () => {
    const wrapper = mount(<CommentBoxForm />);
    wrapper
      .find('button.Form_Button')
      .simulate('click');
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
})


