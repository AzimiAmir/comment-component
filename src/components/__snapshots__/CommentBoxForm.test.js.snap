// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentBoxForm /> should be possible to activate button with click 1`] = `ReactWrapper {}`;
